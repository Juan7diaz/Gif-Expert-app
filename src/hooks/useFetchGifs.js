import { useEffect, useState } from "react"
import getGifs from '../helpers/getGifs'

export const useFetchGifs = ( category, cantGifs ) =>{

    const [state, setState] = useState({
        data: [],
        loading: true
    })
    
    useEffect( ()=>{

        getGifs(category, cantGifs)
            .then( gifs => ( setState({
                data: gifs,
                loading: false
            }
        )))

    },[ category ])


    return state; 
}