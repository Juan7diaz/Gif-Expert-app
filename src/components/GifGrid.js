import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import LoadingComponent from './LoadingComponent';

const GifGrid = ({ category, cantGifs }) => {

    //para renombrar la data es data:nombre_Deseado
    const { data:images, loading } = useFetchGifs( category , cantGifs );
    return (
        <>
            <h3>{ category }</h3>

            {loading &&  <LoadingComponent />}

            <div className="card-grid">

                { /* para cuando  es un array de objetos podemos pasar como props { ...img} */
                    images.map( ( img ) => (
                        <GifGridItem 
                            key={ img.id }
                            /* img={ img }  */
                            { ...img }
                        />
                    ))
                }

            </div>
            
            {( images.length === 0 && loading === false  ) && <p>BÃºsqueda no encontrada ðŸ˜¢</p> }

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid