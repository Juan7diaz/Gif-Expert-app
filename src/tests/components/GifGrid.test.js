import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs'); //para falsear la data, simulamos que el hook ya nos devuelve algo

describe('Pruebas del componente GifGrid', () => {

    const category = 'The Promised Neverland';
    
    test('debe mostrar correctamente el componente ', () => {
        
        //puede ser mockReturn(...)
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category }/>);
        expect( wrapper ).toMatchSnapshot();
    })

    test('should change the snapshot when useFetchGifs is executed', () => {
        
        const data = [{
            id: 'TPN',
            title: 'THE PROMISED NEVERLAND',
            url: 'https://localhost/cualquier/cosa.jpg'
        },
        {
            id: 'WEP',
            title: 'WONDER EGG PRIORITY',
            url: 'https://localhost/cualquier/cosa.jpg'
        }];

        useFetchGifs.mockReturnValue({
            data: data,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);
       
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false ) //para cuando el loading no existe 
        expect( wrapper.find('GifGridItem').length ).toBe( data.length )


    });

})
