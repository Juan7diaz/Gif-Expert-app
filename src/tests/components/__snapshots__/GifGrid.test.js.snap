// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGrid debe mostrar correctamente el componente  1`] = `
<Fragment>
  <h3>
    The Promised Neverland
  </h3>
  <LoadingComponent />
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas del componente GifGrid should change the snapshot when useFetchGifs is executed 1`] = `
<Fragment>
  <h3>
    The Promised Neverland
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="TPN"
      key="TPN"
      title="THE PROMISED NEVERLAND"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="WEP"
      key="WEP"
      title="WONDER EGG PRIORITY"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
